samples/DLLMethodCaller/DLLMethodCallerEA.mq4

```diff
@@ -45,13 +45,20 @@
 int init()
 {
 	nquotes_setup("DLLMethodCaller.DLLMethodCallerEA", "DLLMethodCaller");
-	
-	return (nquotes_init());
+	nquotes_set_property_int ("MoneyAmount",999);
+        nquotes_set_property_string ("ActionName","GiveMoney");
+        nquotes_set_property_datetime ("MyBirthday",D'2000.03.05 15:46:58');
+ 	return (nquotes_init());
 }
 
 int start()
 {
-	return (nquotes_start());
+	 int r=nquotes_start();
+	 //nquotes_set_property_int ("MoneyAmount",MoneyAmount);
+	 nquotes_set_property_string ("ActionName","PredictFuture");
+         string future = nquotes_get_property_string("FuturePrediction");
```

Comment:
You are setting the action and getting prediction here, but not calling nquotes_start() in between. In this case it will not update the prediction, and you are gonna get the previous value.

In order to simplify the MQL code of init and start I'd recommend to make 2 new functions that wrap C# functions in MQL:
1. The first function is `void GiveMoney(int amount)`, that should set parameters for "GiveMoney" and call nquotes_start().
1. The 2nd function is `string PredictFuture()`. that should set parameters for "PredictFuture" and call nquotes_start().

With these functions you can simplify the start() function to call: GiveMoney, PredictFuture and print the result.

If you do everything in start(), you don't need to have nquotes_set_property calls in init.

**Tip**: If you want to test this easily - run this EA as a script. Copy DLLMethodCallerEA.mq4 into "Scripts" folder, and compile from there. Then when you drag it onto the chart (from "scripts"), it will call init() and start() once immediately.